gedditlive/GedditSDK
# Release Event Tracking

# Changelog

- Added: `collection_id` and `collection_name` properties for `product_category_update` event.
- Added: `collection_id` and `collection_name` properties for `like` event.
- Added: `collection_id` and `collection_name` properties for `dismiss_product_list` event.
- Added: `collection_id` and `collection_name` properties for `product_image_pressed` event.
- Added: `collection_id` and `collection_name` properties for `display_product_list` event.
- Added: `collection_id` and `collection_name` properties for `send_chat_message` event.
- Added: `collection_id` and `collection_name` properties for `share` event.
- Added: `show_id` property for `live_screen_tap` event.
- Added: `entrypoint_id` property for `open_app` event.
- Added: `close_sdk` event.
- Added: `add_to_bag` event.
- Removed: `full_screen` property for `display_chat` event.

# Testing

In order to test using this branch, you should edit the cocoapods repository used as a source for your Podfile:

1. Make sure that `~/.cocoapods/repos/gedditlive` exists
    - If not, Run `pod install` with the following Podfile:

```ruby
source 'https://github.com/gedditlive/GedditSDK.git'
target 'YOUR_TARGET' do
  use_frameworks!

  pod 'GedditCoreSDK', '1.1.0'
  pod 'GedditCommerceSDK', '1.2.0'
end
```

2. Switch to the `release/event-tracking` branch:

```sh
cd ~/.cocoapods/repos/gedditlive;
git checkout release/event-tracking
```

3. Edit the podspec files to make `pod install` downloads the frameworks on `release/event-tracking` branch.

_~/.cocoapods/repos/gedditlive/GedditCoreSDK/1.1.0/GedditCoreSDK.podspec:_
```ruby
s.source = { :http => 'https://github.com/gedditlive/GedditSDK/blob/release/event-tracking/GedditCoreSDK/1.1.0/GedditCoreSDK.zip?raw=true'}
```

_~/.cocoapods/repos/gedditlive/GedditCommerceSDK/1.2.0/GedditCommerceSDK.podspec:_
```ruby
s.source = { :http => 'https://github.com/gedditlive/GedditSDK/blob/release/event-tracking/GedditCommerceSDK/1.2.0/GedditCommerceSDK.zip?raw=true'}
```

4. Make sure to clear your pod cache:

```sh
rm -rf ~/Library/Caches/CocoaPods/Pods/Release/GedditCommerceSDK
rm -rf ~/Library/Caches/CocoaPods/Pods/Release/GedditCoreSDK
```

5. Run `pod install` in your project
